### PR TITLE
Base/counter currencies and price inverted for crypto offers

### PR DESCRIPTION
**Pull request by duriancrepe**
Requesting review for haveno-dex/haveno#293. All inversions for crypto (altcoin) offers are removed. 

Removed the base currency and counter currency inversion, then removed the PriceProvider's price inversion for crypto currencies.

Lastly, removed the hack inversion logic for the price triggers and desktop application UI.

There is a bit of awkwardness when creating BUY or SELL orders, because the **MONERO** tab on the desktop application is inconsistent with the new **ETH** or **OTHER CRYPTOCURRENCIES** tab, because you are always buying Monero now, but with a different counter currency.

### CODE REVIEW

**Comment by duriancrepe** [comment]
> Amazing. Thank you.
> 
> The only thing I see is that the buy/sell functionality should be preserved in the legacy UI.
> 
> For example, Buy > ETH should still sell XMR for ETH. Before, that was represented internally as base currency = ETH, counter currency = XMR. Now base currency = XMR and counter currency = ETH, so we should adjust for this in the legacy UI to preserve functionality.

How about the prices in the legacy UI? Do you want the trading pairs and pricing inverted for cryptos like before? For example, XMR / USD shows $200 and then invert it for cryptos like ETH / XMR (17).

**Comment by duriancrepe** [comment]
Restored the legacy UI has the inverted offers creation for cryptos. Prices are displayed with XMR as the base currency.

File: core/src/main/java/bisq/core/locale/CurrencyUtil.java [hunk 1]
```diff
@@ -417,28 +417,19 @@ public static List<CryptoCurrency> getActiveSortedCryptoCurrencies(FilterManager
     }
 
     public static String getCurrencyPair(String currencyCode) {
-        if (isFiatCurrency(currencyCode))
-            return Res.getBaseCurrencyCode() + "/" + currencyCode;
-        else
-            return currencyCode + "/" + Res.getBaseCurrencyCode();
+        return Res.getBaseCurrencyCode() + "/" + currencyCode;
```
[the remark this answers is not in the document]
Comment (duriancrepe) [reply core/src/main/java/bisq/core/locale/CurrencyUtil.java:420]: I can make the change throughout, however isn't it convention that the currency base is displayed first (on the left)? Did some [internet searching](https://www.investopedia.com/terms/c/counter-currency.asp) and that appears to be the case.

**Comment by woodser** [comment]
We've given it further consideration and decided to stick with industry convention.

Convention seems to be displaying price pairs as XMR/fiat and crypto/XMR, and then for fiat, price is per XMR, e.g. $200 per XMR, and for crypto, price is per counter currency, e.g. 0.05 XMR per ETH.

In that case, we don't need this PR (even though it makes sense and simplifies the code). The bounty is still awarded for completion. Thanks and sorry for the lost effort.

**Comment by duriancrepe** [comment]
No problem, its definitely implemented poorly where the abstractions don't make the base / counter currency inversions elegant and require either the isCryptoCurrency / isFiatCurrency utility function sprinkled throughout the application.

The work can still be useful as a reference where the correct abstractions can be made to remove the hacky inversion method (if statements everywhere), but would not be a good use of time and effort at this moment.

**Pull request closed without merging**